pkg/runcfanotify/runcfanotify.go

```diff
@@ -92,9 +104,33 @@ var runcPaths = []string{
 	"/host/run/torcx/unpack/docker/bin/runc",
 }
 
+// true if the SYS_PIDFD_OPEN syscall is available
+var pidfdOpenAvailable bool
+
+func init() {
+	pid := uintptr(os.Getpid())
+	pidfd, _, errno := unix.Syscall(unix.SYS_PIDFD_OPEN, pid, 0, 0)
+	if errno == unix.ENOSYS {
+		log.Debug("SYS_PIDFD_OPEN is not available")
+		pidfdOpenAvailable = false
+		return
+	}
+
+	if errno != 0 {
+		pidfdOpenAvailable = false
```

Comment:
Hum...
I am not convinced by the name of this variable, shouldn't it be `pidfdOpenWorking` or something to reflect this idea?
Because in this case, it is available but not working.

Reply:
I think for practical effects on this code having `SYS_PIDFD_OPEN` available but not working is the same and the `pidfdOpenAvailable` name works fine.